### ISSUE TITLE
package.json scripts run with `yarn run` fail to auth with npm

### ISSUE THREAD

**Issue by mcous** [yarnpkg/yarn#5683]
**Do you want to request a *feature* or report a *bug*?**
bug

**What is the current behavior?**
package.json scripts run with `yarn run` fail to auth with `npm`

**If the current behavior is a bug, please provide the steps to reproduce.**

https://gist.github.com/mcous/0c62847401ed2dc5e230f38746f0c267

0. Ensure you have a valid authToken in `.npmrc`
1. Create a package.json script `"who": "npm whoami"`
2. `npm run who` will work
2. `yarn run who` will fail with `ENEEDAUTH`
    * Workaround: `yarn config set registry https://registry.npmjs.org`
    * Related to #4451?

**What is the expected behavior?**

`package.json` scripts should not fail to auth with `npm` if `.npmrc` is set up correctly

**Please mention your node.js, yarn and operating system version.**

* macOS 10.12.6
* node v8.11.0
* npm v5.8.0
* yarn v1.6.0

**Comment by jamsinclair** [comment]
I believe Yarn doesn't use your local npm directly. It creates an auth token which it passes to its own npm instance to auth itself for commands like, `publish`. 

_(Someone from yarn team feel free to explain more correctly_ 😅 _)_

If you need to run npm commands in yarn scripts that require auth, just login! 😄 
Run `yarn login` on cli, you'll be prompted for your npm username & email. 

--- 

### **tldr;** `yarn login` and then run your scripts

**Comment by jamsinclair** [comment]
Great question! I'm just as unsure. I encountered this problem as well on CI for one of my projects. 

Hopefully a Yarn team member can chime in whether this is a bug or intentional behaviour 🤞

**Comment by itaisteinherz** [comment]
This is still an issue for people (see sindresorhus/np#416), and so it would be great if someone could look into this.

**Comment by jichu4n** [comment]
Ran into this issue as well - when using `yarn run` to execute a script that in turn invokes `npm install`, `npm` is unable to install a private package.

I found another workaround is to add `npm_config_registry=https://registry.npmjs.org` to the script command. For example, in `package.json`:

```
{
    "scripts": {
        "who": "npm_config_registry=https://registry.npmjs.org npm whoami"
    }
}
```

**Comment by sysofwan** [comment]
+1 seeing the same issue. I am trying to invoke publish script from package.json which in turn calls Yarn to publish package to a private registry. Workaround does not really help in this case.

**Comment by austinbutler** [comment]
Workaround from another issue: https://github.com/yarnpkg/yarn/issues/4475#issuecomment-350614480

**Comment by merceyz** [comment]
Closing as in v2 we completely stopped reading `.npmrc` files.

https://yarnpkg.com/getting-started/migration
